app/src/main/java/org/mozilla/tv/firefox/FirefoxApplication.kt

```diff
@@ -50,6 +51,8 @@ open class FirefoxApplication : LocaleAwareApplication() {
         visibilityLifeCycleCallback = VisibilityLifeCycleCallback(this).also {
             registerActivityLifecycleCallbacks(it)
         }
+
+        serviceLocator.fretboardProvider.updateExperiments()
```

Comment:
The chances of our network request finishing before our local storage retrieval are pretty minimal, but I still don't love that this is nondeterministic.  I think I would prefer if `updateExperiments` and `loadExperiments` were suspend funs, and our `onCreate` calls looked something like this

```kotlin
launch(Dispatchers.IO) {
  with(serviceLocator.fretboardProvider) {
    loadExperiments()
    updateExperiments()
  }
}
```

That way `loadExperiments` would block `updateExperiments`, and they would always complete in the same order.

Reply:
Please file this in AC. Happy to make those suspension methods (or add separate ones that suspend).

Other than that: Yes, you'll absolutely want to load before updating since the update call will only request a diff from the server to minimize traffic.

Reply:
Done!

https://github.com/mozilla-mobile/android-components/issues/1855